src/test/kotlin/com/music/awesomemusic/controllers/AccountControllerTest.kt

```diff
@@ -42,64 +46,287 @@ class AccountControllerTest {
+
+        val account =  AccountSignUpForm("Violetta1234", "122334", "email", false)
+        accountService.createAccount(account)
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+
+        val request = HttpEntity<String>("{\"username\": \"VIOLETTA1234\", \"password\": \"1233Agf%%%QWERTY%1qq\",\"email\":\"email@mail.ru\",\"is_collective\": \"false\" }", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/registration", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.CREATED)
+    }
+
+    @Test
+    @Throws(Exception::class)
+    fun shouldSignUpIfLoginExistsWithDifferentRegister2Expected201() {
+
+        val account =  AccountSignUpForm("VIOLETTA1234", "122334", "email", false)
+        accountService.createAccount(account)
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.APPLICATION_JSON
+
+        val request = HttpEntity<String>("{\"username\": \"Violetta1234\", \"password\": \"1233Agf%%%QWERTY%1qq\",\"email\":\"email@mail.ru\",\"is_collective\": \"false\" }", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/registration", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.CREATED)
+    }
+
+
     @Test
     @Throws(Exception::class)
-    fun signInShouldReturnOk() {
+    fun shouldSignUpIfEmailExistsExpected400() {
+
+        val accountSignUpForm = AccountSignUpForm("ViolettaVioletta", "122334", "email@email.ru", false)
+        val account = accountService.createAccount(accountSignUpForm)
+
         val headers = HttpHeaders()
         headers.contentType = MediaType.APPLICATION_JSON
-        val request = HttpEntity<String>("{ \"login\" : \"holker\", \"password\" : \"123456\" }", headers)
-        val response = restTemplate.postForEntity("http://localhost:$port/api/user/sign-in", request,
-                String::class.java)
-        assertEquals(response.statusCode, HttpStatus.OK)
+
+        val request = HttpEntity<String>("{\"username\": \"Violetta\", \"password\": \"1233Agf%%%QWERTY%1qq\",\"email\":\"email@email.ru\",\"is_collective\": \"false\" }", headers)
+        val responseEntity = restTemplate.postForEntity("http://localhost:$port/api/user/registration", request, String::class.java)
+
+        assertEquals(responseEntity.statusCode, HttpStatus.BAD_REQUEST)
     }
 
+
     @Test
-    fun meShouldReturnOk() {
-        val token = tokenProvider.createToken(account.username, emptyList<String>())
+    @Throws(Exception::class)
+    fun shouldMeIfAccountExistsExpected200() {
+
+        val account = accountService.createAccount(AccountSignUpForm("test5", "1234", "emailTest5", false))
+        val authorities = arrayListOf<String>()
+
+        val jwtTokenProvider = tokenProvider.createToken(account.username, authorities)
```

Comment:
```suggestion
        val token = tokenProvider.createToken(account.username, authorities)
```
You should name variable based on what it does